components/mpas-ocean/src/shared/mpas_ocn_diagnostics.F

```diff
@@ -3465,6 +3504,39 @@ subroutine ocn_compute_KPP_input_fields(statePool, forcingPool, &
                       - exp( max(-100.0_RKIND, &
                                  -layerThickness(kmin, iCell)/ &
                            config_flux_attenuation_coefficient_runoff))
+         if (trim(config_subglacial_runoff_mode) == 'data') then
+             if (config_use_sgr_opt_kpp == 1) then
+                if ( trim(config_sgr_flux_vertical_location) == 'top' ) then
+                    fracAbsorbedSubglacialRunoff = 1.0_RKIND &
+                          - exp( max(-100.0_RKIND, &
+                                     -layerThickness(kmin, iCell)/ &
+                               config_flux_attenuation_coefficient_subglacial_runoff))
+                else if ( trim(config_sgr_flux_vertical_location) == 'uniform' ) then
+                   ! calculate total thickness into variable zTop
+                   zTop = 0.0_RKIND
+                   do k = minLevelCell(iCell), maxLevelCell(iCell)
+                      zTop = zTop + layerThickness(k,iCell)
+                   end do
+                   ! distribute flux evenly throughout water column
+                   fracAbsorbedSubglacialRunoff = layerThickness(kmin, iCell) / zTop
+                else if ( trim(config_sgr_flux_vertical_location) == 'bottom' ) then
+                   zTop = 0.0_RKIND
+                   do k = maxLevelCell(iCell), minLevelCell(iCell), -1
+                      zBot = zTop - layerThickness(k,iCell)
+                      if (k == minLevelCell(iCell)) then
+                        transmissionCoeffTop = exp( max(zTop / config_flux_attenuation_coefficient_runoff, -100.0_RKIND) )
+                        transmissionCoeffBot = exp( max(zBot / config_flux_attenuation_coefficient_runoff, -100.0_RKIND) )
+                        fracAbsorbedSubglacialRunoff = transmissionCoeffTop - transmissionCoeffBot
+                      end if
+                      zTop = zBot
+                   end do
+                else
+                    call mpas_log_write("config_sgr_flux_vertical_location not one of 'bottom', 'uniform', 'top'.", MPAS_LOG_CRIT)
```

Comment:
I prefer to have all namelist checks in init routines so that they occur once at the outset of the forward run.

Reply:
We discussed with @xylar and he said in this case it was not necessary but I don't have a strong opinion here.

Reply:
The issue is that there's not currently the need for an init routine so it seems overkill to add it for this one check.

Reply:
You can add it to either this init routine: https://github.com/E3SM-Project/E3SM/blob/7491a836e534a13d4ef1d59ae98b6f36b4f92870/components/mpas-ocean/src/shared/mpas_ocn_diagnostics.F#L4628
or this one: https://github.com/E3SM-Project/E3SM/blob/7491a836e534a13d4ef1d59ae98b6f36b4f92870/components/mpas-ocean/src/shared/mpas_ocn_surface_bulk_forcing.F#L400

Both are called in forward mode init and have all config options available.

Reply:
Ah, yeah, that's a good suggestion.

Reply:
ok, I put it here: ocn_diagnostics_init